src/Tools/ExternalAccess/VisualDiagnostics/Internal/VisualDiagnosticsServiceFactory.cs

```diff
@@ -0,0 +1,103 @@
+/// and handling MAUI XAML/C#/CSS/Razor Hot Reload support
+/// </summary>
+[ExportCSharpVisualBasicLspServiceFactory(typeof(OnInitializedService)), Shared]
+internal sealed class VisualDiagnosticsServiceFactory : ILspServiceFactory
+{
+    private readonly LspWorkspaceRegistrationService _lspWorkspaceRegistrationService;
+    private readonly VisualDiagnosticsServiceBroker _brokeredDebuggerServices;
+
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    [ImportingConstructor]
+    public VisualDiagnosticsServiceFactory(
+        LspWorkspaceRegistrationService lspWorkspaceRegistrationService,
+        VisualDiagnosticsServiceBroker brokeredDebuggerServices)
+    {
+        _lspWorkspaceRegistrationService = lspWorkspaceRegistrationService;
+        _brokeredDebuggerServices = brokeredDebuggerServices;
+    }
+
+    public ILspService CreateILspService(LspServices lspServices, WellKnownLspServerKinds serverKind)
+    {
+        return new OnInitializedService(_lspWorkspaceRegistrationService, _brokeredDebuggerServices);
+    }
+
+    private class OnInitializedService : ILspService, IOnInitialized, IOnServiceBrokerInitialized, IDisposable
+    {
+        private readonly LspWorkspaceRegistrationService _lspWorkspaceRegistrationService;
+        private readonly VisualDiagnosticsServiceBroker _brokeredDebuggerServices;
+        private IVisualDiagnosticsLanguageService? _visualDiagnosticsLanguageService;
+        private CancellationToken _cancellationToken;
+        private static readonly TaskCompletionSource<bool> _taskCompletionSource = new TaskCompletionSource<bool>();
+
+        public OnInitializedService(LspWorkspaceRegistrationService lspWorkspaceRegistrationService, VisualDiagnosticsServiceBroker brokeredDebuggerServices)
+        {
+            _lspWorkspaceRegistrationService = lspWorkspaceRegistrationService;
+            _brokeredDebuggerServices = brokeredDebuggerServices;
+
+            _brokeredDebuggerServices.NotifyServiceBrokerInitialized = this;
+        }
+
+        public void Dispose()
+        {
+            (_visualDiagnosticsLanguageService as IDisposable)?.Dispose();
+        }
+
+        public Task OnInitializedAsync(ClientCapabilities clientCapabilities, RequestContext context, CancellationToken cancellationToken)
+        {
+            _cancellationToken = cancellationToken;
+            _taskCompletionSource.SetResult(true);
+            return Task.CompletedTask;
+        }
+
+        public void OnServiceBrokerInitialized(IServiceBroker serviceBroker)
+        {
+            Task.Run(async () => await OnInitializeVisualDiagnosticsLanguageServiceAsync(serviceBroker).ConfigureAwait(false));
+        }
+
+        private async Task OnInitializeVisualDiagnosticsLanguageServiceAsync(IServiceBroker serviceBroker)
+        {
+            // Make sure we're initialized. 
+            bool initialized = await _taskCompletionSource.Task.ConfigureAwait(false);
```

Comment:
Instead of doing this - in `OnServiceBrokerInitialized` you could do `_taskCompletionSource.Task.ContinueWith(() => OnInitializeVisualDiagnosticsLanguageServiceAsync...`